operator/api/redpanda/v1alpha2/redpanda_types.go

```diff
@@ -355,3 +356,15 @@ func (in *Redpanda) GetDot(restConfig *rest.Config) (*helmette.Dot, error) {
 			IsUpgrade: true,
 		}, in.Spec.ClusterSpec.DeepCopy())
 }
+
+type ClusterWithPools struct {
```

Comment:
Can we move the ClusterWithPools struct into a different package? (Lifecycle seems like the best fit?).

We don't want this to appear in docs (see crd-docs.adoc) and the v1alpha3 package will have to import the v1alpha2 package to implement the conversion interface correct which will cause an import cycle.

Reply:
sure, sounds good